src/print.ts

```diff
@@ -70,10 +70,18 @@ let thisContext: vscode.ExtensionContext;
 export function activate(context: vscode.ExtensionContext) {
     thisContext = context;
     context.subscriptions.push(vscode.commands.registerCommand('markdown.extension.printToHtml', () => { print('html'); }));
+    vscode.workspace.onDidSaveTextDocument(onDidSave);
 }
 
 export function deactivate() { }
 
+function onDidSave(e: vscode.TextDocument) {
+    e.languageId == 'markdown' && 
+        vscode.workspace.getConfiguration(
+            'markdown.extension.printToHtmlOnFileSaved') && 
+            print('html');
+}
```

Comment:
Looks fancy.

However let's make it more formal... (`doc: vscode.TextDocument` and plain `if else`)